### PR TITLE
GAP-2049: Conditionally redirect to summary page after file upload 

### PR DESCRIPTION
**Pull request by john-tco**
## Description

Ticket # and link

Adds logic to redirect an applicant back to the summary page after visiting from there

## Type of change

Please check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes:

- [ ] Unit Test

- [ ] Integration Test (if applicable)

- [ ] End to End Test (if applicable)

## Screenshots (if appropriate):

Please attach screenshots of the change if it is a UI change:

# Checklist:

- [ ] If I have listed dependencies above, I have ensured that they are present in the target branch.
- [ ] I have performed a self-review of my code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation where applicable.
- [ ] I have ran cypress tests and they all pass locally.

### CODE REVIEW

File: packages/applicant/src/pages/api/routes/submissions/[submissionId]/sections/[sectionId]/questions/[questionId]/attachments/[attachmentId]/remove.test.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import handler from './remove.page';
+import { routes } from '../../../../../../../../../../../utils/routes';
+import { deleteAttachmentByQuestionId } from '../../../../../../../../../../../services/SubmissionService';
+
+jest.mock('../../../../../../../../../../../services/SubmissionService');
```
Comment (jgunnCO) [inline packages/applicant/src/pages/api/routes/submissions/[submissionId]/sections/[sectionId]/questions/[questionId]/attachments/[attachmentId]/remove.test.tsx:5]: lol

File: packages/applicant/src/pages/api/routes/submissions/[submissionId]/sections/[sectionId]/questions/[questionId]/attachments/[attachmentId]/remove.test.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      expectedQueryString: '?fromCYAPage=true',
+    },
+    {
+      description: 'with fromSubmissionSummaryPage query string',
+      query: { fromSubmissionSummaryPage: true },
+      expectedQueryString: '?fromSubmissionSummaryPage=true',
+    },
+  ];
+
+  test.each(testCases)(
```
Comment (jgunnCO) [inline packages/applicant/src/pages/api/routes/submissions/[submissionId]/sections/[sectionId]/questions/[questionId]/attachments/[attachmentId]/remove.test.tsx:29]: convention is to use `it` in this repo

```suggestion
  it.each(testCases)(
```

File: packages/applicant/src/pages/submissions/[submissionId]/sections/[sectionId]/questions/[questionId]/index.page.tsx [hunk 3]
```diff
@@ -196,6 +196,11 @@ export default function QuestionPage({
     fieldErrors: error || [],
     titleSize: titleSizeType,
   };
+
+  const fromSubmissionSummaryPage =
+    new URLSearchParams(queryParams).get('fromSubmissionSummaryPage') ===
+    'true';
```
Comment (jgunnCO) [inline packages/applicant/src/pages/submissions/[submissionId]/sections/[sectionId]/questions/[questionId]/index.page.tsx:202]: ??? did prettier enforce this?

Comment (john-tco) [reply packages/applicant/src/pages/submissions/[submissionId]/sections/[sectionId]/questions/[questionId]/index.page.tsx:202]: It does indeed

**Review by jgunnCO — APPROVED** [review]

**Pull request merged**